api/src/main/java/org/apache/iceberg/transforms/TimestampTransform.java

```diff
@@ -28,22 +28,47 @@
 import org.apache.iceberg.expressions.Expression;
 import org.apache.iceberg.expressions.Expressions;
 import org.apache.iceberg.expressions.UnboundPredicate;
+import org.apache.iceberg.relocated.com.google.common.base.Objects;
 import org.apache.iceberg.types.Type;
 import org.apache.iceberg.types.Types;
 
-enum Timestamps implements Transform<Long, Integer> {
-  YEAR(ChronoUnit.YEARS, "year"),
-  MONTH(ChronoUnit.MONTHS, "month"),
-  DAY(ChronoUnit.DAYS, "day"),
-  HOUR(ChronoUnit.HOURS, "hour");
+abstract class TimestampTransform implements Transform<Long, Integer> {
 
   private static final OffsetDateTime EPOCH = Instant.ofEpochSecond(0).atOffset(ZoneOffset.UTC);
-  private final ChronoUnit granularity;
-  private final String name;
 
-  Timestamps(ChronoUnit granularity, String name) {
+  private ChronoUnit granularity;
+  private String name;
+  private ZoneOffset zoneOffset;
+
+  @SuppressWarnings("unchecked")
+  static TimestampTransform get(Type type, String name) {
+    if (type.typeId() == Type.TypeID.TIMESTAMP) {
+      switch (name.toUpperCase()) {
+        case "YEAR":
+          return new TimestampTransform.TimestampYear(name.toLowerCase());
```

Comment:
It's strange to me that this converts name to both upper and lower case. Why not convert to lower case once, use the variable in the `switch` statement, and pass the variable here as well?

Reply:
Also, this should use `Locale.ENGLISH` or `Locale.ROOT` instead of the default locale.